src/Makefile

```diff
@@ -248,7 +248,7 @@ EMPTY:=
 SPACE:=$(EMPTY) $(EMPTY)
 CLANG_TIDY_HEADER_FILTER=$(PWD)/($(subst $(SPACE),|,$(LINTER_HEADERS)))
 
-CXXFLAGS+=$(OPTFLAGS) -std=c++17 -fvisibility=hidden -fPIC -MMD $(CC_MARCH) $(INCS) $(GCFLAGS) $(UBFLAGS) $(DEFS) $(WARNS)
+CXXFLAGS+=$(OPTFLAGS) -std=gnu++17 -fvisibility=hidden -fPIC -MMD $(CC_MARCH) $(INCS) $(GCFLAGS) $(UBFLAGS) $(DEFS) $(WARNS)
```

Comment:
What crazyness required this?

Reply:
Some systems that I cross compiled (such as DOS), did not have all POSIX functions available. Using just `-std=c++17` did not enable non standard C OS APIs, such as `gettimeofday`, `mkdir`, and others, because they are not in standard C. I think some systems are free to disable POSIX APIs with `-std=c++17`.

So having `-std=gnu++17` is more friendly for cross compilation, another option could be removing `-std=` and let use the default one.

Reply:
In an ideal world, we would change this flag only on os.cpp, right?